src/gui/tray/usermodel.cpp

```diff
@@ -1630,10 +1659,11 @@ void UserAppsModel::buildAppList()
 
     if (UserModel::instance()->appList().count() > 0) {
         const auto talkApp = UserModel::instance()->currentUser()->talkApp();
-        foreach (AccountApp *app, UserModel::instance()->appList()) {
+        for (auto &app : UserModel::instance()->appList()) {
             // Filter out Talk because we have a dedicated button for it
-            if (talkApp && app->id() == talkApp->id())
+            if (talkApp && app->id() == talkApp->id() && !UserModel::instance()->currentUser()->isNcAssistantEnabled()) {
```

Comment:
I am really curious why the original code is getting a pointer to talk app from `UserModel::instance()->appList()` and instead of comparing the pointer, it uses the `id()`
maybe you will want to rationalize this

Reply:
let's close this as it is somehow unrelated to your changes
let's for history keep a note that I do not understand the original code
I especially do not understand why the talkApp pointer is getting queried via `UserModel::instance()->currentUser()` to be used when accessing a list of applications queried via `UserModel::instance()`
my problem is that the list of server applications has to be linked to the account and it seems like our current API and use of it is not doing this